tests/test_core/test_new_alignment.py

```diff
@@ -1982,6 +1957,20 @@ def test_sequence_collection_rename_seqs():
     new = seqs.rename_seqs(lambda x: x.upper())
     expect = {n.upper() for n in data}
     assert set(new.names) == expect
+    # the names should not change in the seqsdata
+    assert set(new._seqs_data.names) == set(data)
+
+
+def test_sequence_collection_subsequent_rename():
+    """sequences can be renamed multiple times"""
+    data = {"seq1": "ACGTACGTA", "seq2": "ACCGAA---", "seq3": "ACGTACGTT"}
+    seqs = new_alignment.make_unaligned_seqs(data, moltype="dna")
+    new = seqs.rename_seqs(lambda x: x.upper())
+    new_again = new.rename_seqs(lambda x: x[:1])
+    expect = {"S"}
```

Comment:
this makes me realise the rename_seqs function should check that the resulting names are unique, otherwise a user goes from 3 seqs down to 1.

please add a test that fails with ValueError if the number of names in the resulting collection is not equal to the number in the original (see my suggestion above).

for this test, change the second lambda to
```python
new_again = new.rename_seqs(lambda x: f"{x[0]}{x[-1]}")
```